src/coreclr/jit/morph.cpp

```diff
@@ -11504,6 +11502,13 @@ GenTree* Compiler::fgPropagateCommaThrow(GenTree* parent, GenTreeOp* commaThrow,
     assert(fgGlobalMorph);
     assert(fgIsCommaThrow(commaThrow));
 
+    bool mightBeLocation = (parent->OperIsLocal() || parent->OperIsIndir()) && ((parent->gtFlags & GTF_DONT_CSE) != 0);
```

Comment:
`parent->OperIsLocal` looks a bit unnecessary as `LCL_VAR/LCL_FLD` LHSs cannot have children, and `STORE_LCL_VAR/FLD` (when those start appearing here) won't need to be restricted like this.

Reply:
Removed that check.